### ISSUE TITLE
improved yolov8 model 

### ISSUE THREAD

**Issue by emre888** [ultralytics/ultralytics#1406]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, I made some changes on original yolov8s.yaml file. I added attention model.  How will I train this model by using pretrained coco yolov8s.pt file. can yolov8 transfer same layers' weights

**Comment by glenn-jocher** [comment]
@emre888 this isn't possible right now, but multiple users have inquired about this, so we may add it to our feature roadmap.

**Comment by emre888** [comment]
I would be grateful because training from scratch for just a few changes on the original model is time consuming and I couldnt get the same success. Moreover, this v8 version goes up to very high mAPs even after the first few epochs on pretraining mode,, unlike v5,v7

**Comment by emre888** [comment]
@glenn-jocher I would be grateful because training from scratch for just a few changes on the original model is time consuming and I couldnt get the same success. Moreover, this v8 version goes up to very high mAPs even after the first few epochs on pretraining mode, unlike v5,v7

**Comment by glenn-jocher** [comment]
@emre888 oh interesting, good to know. Yes this is because the YOLOv8 objectness is gone, we've removed that loss, which was targeted to IoU, which in early training is always low, so mAP takes longer to ramp up in v5 and v7 (which is based mostly on the v5 repo).

**Comment by glenn-jocher** [comment]
@emre888 I'm thinking for Python we can simply accept two arguments, i.e.:

`model = YOLO('yolov8n.pt', 'custom.yaml')` in any order, but for CLI I'm not sure what argument to use, and I don't want to add additional arguments because we already have so many. What about a list of arguments so you can pass a yaml and a pt?

```
yolo train model=yolov8n.pt,custom.yaml
# or
yolo train model=yolov8n.pt NEW_ARG=custom.yaml
```
@AyushExel @Laughing-q what do you guys think?

**Comment by emre888** [comment]
@glenn-jocher  there is 'pretrained' bool arg, I dont know if it has effect on detection mode. 
yolo train model=custom.yaml,yolov8n.pt     pt1,pt2   Could this appoach work for ensemble weights,too?

**Comment by urbaneman** [comment]
I also have the same requirement. 

In the current version, the `model=yolov8s.pt` loads the model structure directly from the .pt file, and any YAML file that conflicts with the current model file will be invalidated. 

https://github.com/ultralytics/ultralytics/blob/30fc4b537ff1d9b115bc1558884f6bc2696a282c/ultralytics/yolo/engine/trainer.py#L417-L432

Additionally, I have noticed that the `pretrained` field is not being utilized effectively in the `detect` task. Perhaps this field can be used in the `detect` task to load a pre-trained model.

**Comment by AyushExel** [comment]
yes we'll use pretrained for detect and segment task too
